### PR TITLE
Add custom site template

### PR DESCRIPTION
**Pull request by speedygamer12**
Add custom spider template using start URL

### CODE REVIEW

File: templates/start_url/{{cookiecutter.spider_name}}/{{cookiecutter.spider_name|lower}}.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+from scrapy.http.response import Response
+
+from ._base import ArticleData
+from ._base import CustomSpider
+
+
+logger = logging.getLogger(__name__)
+
+
+class {{cookiecutter.spider_name|capitalize}}Spider(CustomSpider):
```
Comment (theyorubayesian) [inline templates/start_url/{{cookiecutter.spider_name}}/{{cookiecutter.spider_name|lower}}.py]: Can you include an if use_selenium block?

File: templates/start_url/{{cookiecutter.spider_name}}/{{cookiecutter.spider_name|lower}}.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+start_urls:
+  -
+
```
Comment (theyorubayesian) [inline templates/start_url/{{cookiecutter.spider_name}}/{{cookiecutter.spider_name|lower}}.yaml]: Can you include an if use_selenium block?

**Comment by theyorubayesian** [comment]
Can you rename the template? Maybe `next_page_navigation` instead of `start_url` since this particular spider works by finding the next page until all articles are collected.

**Comment by speedygamer12** [comment]
Yh I agree that next_page_navigation looks better. I initially used start_url because we require a start_url  in this case, just like a sitemap for the other.

**Comment by theyorubayesian** [comment]
Also, looks like having `|` in the filenames causes an issue on Windows because it's a reserved character. See #12

File: templates/next_page_navigation/{{cookiecutter.spider_name}}/{{cookiecutter.spider_name}}.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+from ._base import ArticleData
+from ._base import CustomSpider
+
+
+logger = logging.getLogger(__name__)
+
+
+class {{cookiecutter.spider_name|capitalize}}Spider(CustomSpider):
+    name = "{{ cookiecutter.spider_name|lower }}_spider"
```
Comment (theyorubayesian) [inline templates/next_page_navigation/{{cookiecutter.spider_name}}/{{cookiecutter.spider_name}}.py:26]: there needs to be a if_selenium block here. See [L20-L22](https://github.com/theyorubayesian/otelemuye/blob/asiwaju/templates/sitemap/%7B%7Bcookiecutter.spider_name%7D%7D/%7B%7Bcookiecutter.spider_name%7D%7D.py#L20-L22)

**Review by theyorubayesian — CHANGES_REQUESTED** [review]

**Review by theyorubayesian — APPROVED** [review]

**Review by theyorubayesian — APPROVED** [review]

**Pull request merged**